src/dbnode/client/connection_pool.go

```diff
@@ -182,30 +182,25 @@ func (p *connPool) connectEvery(interval time.Duration, stutter time.Duration) {
 		var wg sync.WaitGroup
 		for i := 0; i < target-poolLen; i++ {
 			wg.Add(1)
-			newConnFn := p.opts.NewConnectionFn()
 			go func() {
 				defer wg.Done()
 
-				// Create connection
-				channel, client, err := newConnFn(channelName, address, p.opts)
+				ch, client, healthCheckErr, err := establishNewConnection(address, false, p.opts)
 				if err != nil {
-					log.Debug("could not connect", zap.String("host", address), zap.Error(err))
-					return
-				}
-
-				// Health check the connection
-				if err := p.healthCheckNewConn(client, p.opts, false); err != nil {
-					p.maybeEmitHealthStatus(healthStatusCheckFailed)
-					log.Debug("could not connect, failed health check", zap.String("host", address), zap.Error(err))
-					channel.Close()
+					if healthCheckErr {
+						p.maybeEmitHealthStatus(healthStatusCheckFailed)
+					}
+					log.Warn("could not connect", zap.String("host", address), zap.Error(err))
 					return
 				}
 
 				p.maybeEmitHealthStatus(healthStatusOK)
 				p.Lock()
 				if p.status == statusOpen {
-					p.pool = append(p.pool, conn{channel, client})
+					p.pool = append(p.pool, conn{ch, client})
 					p.poolLen = int64(len(p.pool))
+				} else {
+					ch.Close()
```

Comment:
This is likely a corner case since pool is being closed on shutdown only. Just trying to be a good citizen and cleanup. Also did not figure an easy way to test this.